src/extension/importer/core/src/main/java/org/geoserver/importer/FeatureDataConverter.java

```diff
@@ -225,42 +279,54 @@ public SimpleFeatureType convertType(
+                    // See ensureOracleSafe for conversion
+
                     Set<String> fromAttrNames = attributeNames(from);
                     Set<String> toAttrNames = attributeNames(to);
+
                     for (String name : fromAttrNames) {
-                        String toName = name.toUpperCase();
+                        String toName = ensureOracleSafe(name);
                         if (toAttrNames.contains(toName)) {
                             to.setAttribute(convertAttributeName(toName), from.getAttribute(name));
                         }
                     }
-                };
+                }
 
-                @Override
-                public SimpleFeatureType convertType(
-                        SimpleFeatureType featureType,
-                        VectorFormat format,
-                        ImportData data,
-                        ImportTask task) {
-                    SimpleFeatureTypeBuilder typeBuilder = new SimpleFeatureTypeBuilder();
-                    AttributeTypeBuilder attributeBuilder = new AttributeTypeBuilder();
-                    typeBuilder.setName(ensureOracleSafe(featureType.getTypeName()));
-                    for (AttributeDescriptor att : featureType.getAttributeDescriptors()) {
-                        attributeBuilder.init(att);
-                        final String name = (ensureOracleSafe(att.getName().getLocalPart()));
-                        typeBuilder.add(attributeBuilder.buildDescriptor(name));
-                    }
-                    return typeBuilder.buildFeatureType();
+                /**
+                 * Override to check that typeName is {@link #ensureOracleSafe(String)}.
+                 *
+                 * @param typeName
+                 * @return type name, ensuring this is a valid table name for oracle
+                 */
+                protected String convertTypeName(String typeName) {
+                    typeName = super.convertTypeName(typeName);
+                    return ensureOracleSafe(typeName);
                 }
 
+                /**
+                 * Override to ensure attribute name is {@link #ensureOracleSafe(String)}.
+                 *
+                 * @param attName proposed attribute name
+                 * @return attribtue name, ensuring this is a valid column name for oracle
+                 */
+                protected String convertAttributeName(String attName) {
+                    attName = super.convertAttributeName(attName);
+
+                    return ensureOracleSafe(attName);
+                }
+                /**
+                 * Capitalize and ensure identifier does not conflict with Oracle reserved words.
+                 *
+                 * @param identifier Proposed table or column name
+                 * @return identifier for use as table or column name
+                 */
                 private final String ensureOracleSafe(final String identifier) {
```

Comment:
Ideally, this should use the SQLDialect methods from your oracle datastore.

Reply:
yes, but am concerned about failure at runtime